src/graph6.rs

```diff
@@ -0,0 +1,995 @@
+        Ok(Ok(ParserResult::Graph(g))) => graph_to_pygraph(py, &g),
+        Ok(Ok(ParserResult::DiGraph(dg))) => digraph_to_pydigraph(py, &dg),
+        Ok(Err(io_err)) => Err(PyErr::from(io_err)),
+        Err(panic_payload) => {
+            let msg = if let Some(s) = panic_payload.downcast_ref::<&str>() {
+                format!("Rust panic in graph6 parser: {}", s)
+            } else if let Some(s) = panic_payload.downcast_ref::<String>() {
+                format!("Rust panic in graph6 parser: {}", s)
+            } else {
+                "Rust panic in graph6 parser (non-string payload)".to_string()
+            };
+            Err(Graph6PanicError::new_err(msg))
+        }
+    }
+}
+
+#[pyfunction]
+#[pyo3(signature=(pygraph))]
+pub fn write_graph6_from_pygraph(pygraph: Py<PyGraph>) -> PyResult<String> {
+    Python::with_gil(|py| {
+        let g = pygraph.borrow(py);
+        let n = g.graph.node_count();
+        if n >= (1usize << 36) {
+            return Err(Graph6OverflowError::new_err("Graph too large for graph6 encoding"));
+        }
+        // build bit_vec
+        let mut bit_vec = vec![0usize; n * n];
+        for (i, j, _w) in get_edge_iter_with_weights(&g.graph) {
+            bit_vec[i * n + j] = 1;
+            bit_vec[j * n + i] = 1;
+        }
+        let graph6 = write::write_graph6(bit_vec, n, false);
+        Ok(graph6)
+    })
+}
+
+/// Parse the size header of a graph6 or digraph6 string.
+///
+/// Returns a tuple (n, size_field_length). For a directed (digraph6) string
+/// starting with '&', pass offset=1. This function enforces canonical
+/// encoding (shortest valid length) per the specification in:
+/// https://users.cecs.anu.edu.au/~bdm/data/formats.txt
+#[pyfunction]
+#[pyo3(signature=(data, offset=0))]
+pub fn parse_graph6_size(data: &str, offset: usize) -> PyResult<(usize, usize)> {
+    let bytes = data.as_bytes();
+    let (n, consumed) = utils::parse_size(bytes, offset)?;
+    Ok((n, consumed))
+}
+
+
+/// Read a graph6 file from disk and return a PyGraph or PyDiGraph
+#[pyfunction]
+#[pyo3(signature=(path))]
+pub fn read_graph6_file<'py>(py: Python<'py>, path: &str) -> PyResult<Bound<'py, PyAny>> {
+    use std::fs;
+    let data = fs::read_to_string(path)
+        .map_err(|e| PyErr::new::<pyo3::exceptions::PyIOError, _>(format!("IO error: {}", e)))?;
+    // graph6 files may contain newlines; take first non-empty line
+    let line = data.lines().find(|l| !l.trim().is_empty()).unwrap_or("");
+    read_graph6_str(py, line)
+}
+
+/// Write a PyGraph to a graph6 file
+#[pyfunction]
+#[pyo3(signature=(graph, path))]
+pub fn graph_write_graph6_file(graph: Py<PyGraph>, path: &str) -> PyResult<()> {
+    let s = write_graph6_from_pygraph(graph)?;
+    to_file(path, &s)
+        .map_err(|e| PyErr::new::<pyo3::exceptions::PyIOError, _>(format!("IO error: {}", e)))?;
+    Ok(())
+}
+
+/// Write a PyDiGraph to a graph6 file
+// digraph write helpers are provided by crate::digraph6
+
+#[cfg(test)]
```

Comment:
Either migrate these tests to the Python unit tests or remove them.`cargo test` will not run those.